addons/binding/org.openhab.binding.tankerkoenig/src/main/java/org/openhab/binding/tankerkoenig/handler/BridgeHandler.java

```diff
@@ -0,0 +1,512 @@
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+
+import org.joda.time.DateTime;
+import org.joda.time.LocalDate;
+import org.joda.time.LocalTime;
+import org.openhab.binding.tankerkoenig.TankerkoenigBindingConstants;
+import org.openhab.binding.tankerkoenig.internal.config.LittleStation;
+import org.openhab.binding.tankerkoenig.internal.config.OpeningTime;
+import org.openhab.binding.tankerkoenig.internal.config.OpeningTimes;
+
+import org.openhab.binding.tankerkoenig.internal.config.TankerkoenigListResult;
+import org.openhab.binding.tankerkoenig.internal.data.TankerkoenigService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BridgeHandler} is responsible for handling the things (Tankstellen)
+ *
+ *
+ * @author Dennis Dollinger/Jürgen Baginski
+ */
+public class BridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private String apiKey;
+    private int refreshInterval;
+    private boolean setupMode;
+    private boolean use_OpeningTime;
+
+    private ArrayList<Thing> tankstellenThingList;
+    private HashMap<String, LittleStation> tankstellenList;
+
+    TankerkoenigListResult tankerkoenigListResult;
+
+    private ScheduledFuture<?> pollingJob;
+
+    public BridgeHandler(Bridge bridge) {
+        super(bridge);
+        this.tankstellenThingList = new ArrayList<Thing>();
+        tankstellenList = new HashMap<String, LittleStation>();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // no code needed.
+    }
+
+    @Override
+    public void initialize() {
+        Configuration config = getThing().getConfiguration();
+        this.setApiKey((String) config.get(TankerkoenigBindingConstants.CONFIG_API_KEY));
+        this.setRefreshInterval(((BigDecimal) config.get(TankerkoenigBindingConstants.CONFIG_REFRESH)).intValue());
+        this.setSetupMode((boolean) config.get(TankerkoenigBindingConstants.CONFIG_SETUP_MODE));
+        this.setUseOpeningTime((boolean) config.get(TankerkoenigBindingConstants.CONFIG_USE_OPENINGTIME));
+
+        updateStatus(ThingStatus.ONLINE);
+
+        int pollingPeriod = this.getRefreshInterval();
+        pollingJob = scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                logger.debug("Try to refresh data");
+                // Just update data if setupMode is false(off)
+                try {
+                    if (!isSetupMode()) {
+                        updateTankstellenData();
+                        updateTankstellenThings();
+                    }
+                } catch (Throwable t) {
+                    logger.error("Caught exception in ScheduledExecutorService of BridgeHandler. StackTrace: {}",
```

Comment:
You do not need to convert the stack trace yourself, just call logger.error("blabla", t)